packages/playwright-ct-core/src/vitePlugin.ts

```diff
@@ -198,20 +189,22 @@ export function createPlugin(
         log('build');
         await build(finalConfig);
         await fs.promises.rename(`${finalConfig.build.outDir}/${relativeTemplateDir}/index.html`, `${finalConfig.build.outDir}/index.html`);
+        buildInfo.deps = Object.fromEntries(depsCollector.entries());
       }
 
-      if (hasNewTests || hasNewComponents || sourcesDirty) {
+      if (hasNewComponents || sourcesDirty) {
         log('write manifest');
         await fs.promises.writeFile(buildInfoFile, JSON.stringify(buildInfo, undefined, 2));
       }
 
-      for (const [filename, testInfo] of Object.entries(buildInfo.tests)) {
+      for (const [testFile, componentFiles] of componentsByTestFile) {
         const deps = new Set<string>();
-        for (const componentName of testInfo.components) {
-          const component = componentRegistry.get(componentName);
-          component?.deps.forEach(d => deps.add(d));
+        for (const componentFile of componentFiles) {
+          const componentDeps = depsCollector.get(componentFile) || [];
+          for (const d of componentDeps)
+            deps.add(d);
         }
-        setExternalDependencies(filename, [...deps]);
+        setExternalDependencies(testFile, [...deps]);
```

Comment:
I guess I am not following something, but to me it seems that `testFile` here could be a dependency, and I'm not really sure whether `setExternalDependencies()` for a non-test file ever has any effect on the system.

Reply:
Renaming, updating!